src/ext/service.rs

```diff
@@ -55,6 +60,18 @@ impl ServiceExt for dc::Service {
         Ok(srcdir.value()?.to_owned())
     }
 
+    fn repo_subdir(&self) -> Result<Option<String>> {
+        if let Some(context) = self.context()? {
+            match *context {
+                dc::Context::Dir(_) => (),
+                dc::Context::GitUrl(ref git_url) => {
+                    return Ok(git_url.subdirectory())
```

Comment:
Again, new method from the other PR.

Reply:
The `dc::Context::Dir(_) => (),` idiom feels a bit odd. Maybe:

```rust
                dc::Context::Dir(_) => Ok(none),
                dc::Context::GitUrl(ref git_url) => {
                    Ok(git_url.subdirectory())
```

Reply:
Yeah I was trying to avoid repeating the `Ok(None)` but I agree that it's weird this way.